support-frontend/assets/pages/[countryGroupId]/components/checkoutComponent.tsx

```diff
@@ -159,30 +161,34 @@ type CreateSubscriptionResponse = StatusResponse & {
 	userType: UserType;
 };
 
-const processPayment = async (
+const processPaymentWithRetries = async (
 	statusResponse: StatusResponse,
-	geoId: GeoId,
 ): Promise<ProcessPaymentResponse> => {
-	return new Promise((resolve) => {
-		const { trackingUri, status, failureReason } = statusResponse;
-		if (status === 'success') {
-			resolve({ status: 'success' });
-		} else if (status === 'failure') {
-			resolve({ status, failureReason });
-		} else {
-			setTimeout(() => {
-				void fetch(trackingUri, {
-					headers: {
-						'Content-Type': 'application/json',
-					},
-				})
-					.then((response) => response.json())
-					.then((json) => {
-						resolve(processPayment(json as StatusResponse, geoId));
-					});
-			}, 1000);
-		}
-	});
+	const { trackingUri, status } = statusResponse;
+	if (status === 'success' || status === 'failure') {
+		return processPayment(statusResponse);
+	}
+	const getTrackingStatus = () =>
+		fetch(trackingUri, {
+			headers: {
+				'Content-Type': 'application/json',
+			},
+		}).then((response) => response.json() as unknown as StatusResponse);
+
+	return retryPaymentStatus(getTrackingStatus).then((response) =>
+		processPayment(response),
+	);
+};
+
+const processPayment = (
```

Comment:
Very nitpicky but I wonder if this should be called `handlePaymentStatus` or similar? It's not really processing anything more just mapping the response to something we can use here.

I also wonder if the return value should just be a `ProcessPaymentResponse`, do we need to wrap both branches of the if/else in a `Promise`?

Reply:
Thats a good spot, have updated this and removed the promise